universal-application-tool-0.0.1/app/filters/LoggingFilter.java

```diff
@@ -37,6 +45,31 @@ public EssentialAction apply(EssentialAction next) {
                     long time = clock.millis() - startTime;
                     log.info(
                         "{}\t{}\t{}ms\t{}", request.method(), request.uri(), time, result.status());
+                    StringBuilder requestCookies = new StringBuilder();
+                    Iterator<Http.Cookie> requestCookieIterator = request.cookies().iterator();
+                    while (requestCookieIterator.hasNext()) {
+                      Http.Cookie cookie = requestCookieIterator.next();
+                      requestCookies.append(
+                          String.format("key: %s, value: %s\n", cookie.name(), cookie.value()));
+                    }
+                    log.trace("request cookies: {}", requestCookies.toString());
+                    log.trace("response headers: {}", result.headers().toString());
+                    if (result.body() instanceof HttpEntity.Strict) {
+                      log.trace(
+                          "response body: {}",
+                          ((HttpEntity.Strict) result.body())
+                              .data()
+                              .decodeString(StandardCharsets.UTF_8));
+                    }
+
+                    StringBuilder responseCookies = new StringBuilder();
+                    Iterator<Http.Cookie> responseCookieIterator = result.cookies().iterator();
+                    while (responseCookieIterator.hasNext()) {
+                      Http.Cookie cookie = responseCookieIterator.next();
+                      responseCookies.append(
+                          String.format("key: %s, value: %s\n", cookie.name(), cookie.value()));
+                    }
```

Comment:
suggestion (non-blocking): It might be more readable to pull out the cookie-logging logic into a separate method and just pass in the cookies to log

Reply:
This is so annoying but actually they are *different* cookie objects.  Cookies vs Http.Cookie.  :(